http_types/types.py

```diff
@@ -40,6 +41,11 @@ class _Request(TypedDict, total=False):
     """
     bodyAsJson: Any
 
+    """
+    Timestamp when the request was initiated.
+    """
+    timestamp: Optional[datetime]
```

Comment:
The way I've added optional fields in TypedDicts is by adding them to the base class like `_Request`, that one has `total=False` which means they're "optional". I think it's better to stick with that convention for now instead of using `Optional` as field type?

Reply:
Your approach would probably work better for run-time typechecking (using libraries like `typeguard`) because the base class information is not available anymore at run-time (https://github.com/agronholm/typeguard/issues/101). But I think static type checking is more important  and that can be done by mixing `total=True` and `total=False` classes like here https://mypy.readthedocs.io/en/stable/more_types.html#mixing-required-and-non-required-items

Reply:
If we don't have `Optional` here, we get the following error trying to assign `None` to the `timestamp`:

> TypeError: type of dict item "timestamp" for request must be datetime.datetime; got NoneType instead

Reply:
Sorry I was confused by the GitHub diff, it's in the right place! Sorry I don't understand why you would assign `None` to `timestamp` if `timestamp` is not a required field. Is that a problem in deserialization somehow?

Reply:
Same as in https://github.com/Meeshkan/py-http-types/pull/14#discussion_r373999032.